akka-discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -18,6 +18,14 @@ import akka.util.HashCode
 
 object ServiceDiscovery {
 
+  /**
+   * Future returned by resolve(name, timeout) should be failed with this exception
+   * if the underlying mechanism was unable to resolve the name within the given timeout.
+   *
+   * It is up to each implementation to implement timeouts.
+   */
+  final class DiscoveryTimeoutException() extends Exception
```

Comment:
Wouldn't it be good to allow a message as well, or is the cause of the timeout always obvious?

Reply:
I guess so, I couldn't think of anything apart from saying timed out :)

Reply:
Even if we don't have anything to say here, some other discovery implicit could perhaps have several parts/steps that could time out where a desc would be useful (I'm not sure here, just guessing)?